compiler/rustc_privacy/src/lib.rs

```diff
@@ -775,9 +820,19 @@ impl<'tcx> Visitor<'tcx> for EmbargoVisitor<'tcx> {
             hir::ItemKind::Struct(ref struct_def, _) | hir::ItemKind::Union(ref struct_def, _) => {
                 if item_level.is_some() {
                     self.reach(item.def_id, item_level).generics().predicates();
+
+                    if let Some(ctor_id) = struct_def.ctor_hir_id() {
```

Comment:
The changes here and few lines below look unnecessary, these updates are already done in the "Update levels of nested things" block.